### PR TITLE
Vacancy edit and archive forms

### LINKED ISSUE #471: Vacancy edit and archive forms
In VacancyListItemEditable add edit and archive forms to DropdownMenu

Disallow editing of the hiring stream and the team where the vacancy was created.

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #471 

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/components/ArchiveVacancyModal/ArchiveVacancyModal.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    vacancy: Vacancy;
+    onClose: VoidFunction;
+};
+
+export const ArchiveVacancyModal = ({ visible, vacancy, onClose }: ArchiveVacancyModalProps) => {
+    const { archiveVacancy } = useVacancyMutations();
+
+    const onArchiveClick = async () => {
+        await archiveVacancy(vacancy.id);
+        onClose();
```
Comment (9teen90nine) [inline src/components/ArchiveVacancyModal/ArchiveVacancyModal.tsx:29]: Doesn't it close on error?

Comment (nyet-ty) [reply src/components/ArchiveVacancyModal/ArchiveVacancyModal.tsx:29]: No, it doesn't

**Review by 9teen90nine — APPROVED** [review]

File: src/components/RoleComboBox/RoleComboBox.tsx [hunk 2]
```diff
@@ -12,11 +12,12 @@ import { tr } from './RoleComboBox.i18n';
 
 interface RoleComboBoxProps {
     role?: Nullish<Role>;
+    roleName?: string;
```
Comment (tsumo) [inline src/components/RoleComboBox/RoleComboBox.tsx:15]: What's this?

**Review by tsumo — CHANGES_REQUESTED** [review]

File: src/components/RoleComboBox/RoleComboBox.tsx (hunk 2 above)
Comment (nyet-ty) [reply src/components/RoleComboBox/RoleComboBox.tsx:15]: This for vacancy name: it doesn't have link to role, so it has only role name

**Review by tsumo — APPROVED** [review]

**Pull request merged**